modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/AbstractJavaExtensionModelLoader.java

```diff
@@ -119,16 +104,38 @@ protected void declareExtension(ExtensionLoadingContext context) {
     delegateFactory.apply(extensionType, version).declare(context);
   }
 
+  private Collection<DeclarationEnricher> getPrivilegedDeclarationEnrichers(ExtensionLoadingContext context) {
+    Class<?> extensionType = getExtensionType(context);
+    try {
+      // TODO: MULE-12744. If this call throws an exception it means that the extension cannot access the privileged API.
+      Class annotation = context.getExtensionClassLoader().loadClass(DeclarationEnrichers.class.getName());
+      DeclarationEnrichers enrichers = extensionType.getAnnotation((Class<DeclarationEnrichers>) annotation);
+      if (enrichers != null) {
+        return stream(enrichers.value()).map(this::instantiateOrFail).collect(toList());
```

Comment:
you overlooked my previous comment. This line needs to happen with context.getExtensionClassLoader() as TCCL